markdown_it/extensions/tasklists/__init__.py

```diff
@@ -0,0 +1,159 @@
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+from math import ceil
+from random import random
+from typing import List
+
+from markdown_it import MarkdownIt
+
+from markdown_it.token import Token
+
+
+def tasklists_plugin(
+    md: MarkdownIt,
+    enabled: bool = False,
+    label: bool = False,
+    label_after: bool = False,
+):
+    """Plugin for building task/todo lists out of markdown lists with items starting with [ ] or [x]
+    .. Nothing else
+
+    The rendered checkboxes are disabled; to change this, pass a truthy value into the enabled
+    property of the plugin options.
+
+    :param enabled: True enables the rendered checkboxes
+    :param label: True wraps the rendered list items in a <label> element for UX purposes,
+    :param label_after: True – adds the <label> element after the checkbox.
+    """
+    disable_checkboxes = not enabled
+    use_label_wrapper = label
+    use_label_after = label_after
+
+    def fcn(state):
+        tokens: List[Token] = state.tokens
+        for i in range(2, len(tokens) - 1):
+
+            if is_todo_item(tokens, i):
+                todoify(tokens[i], tokens[i].__class__)
+                attr_set(
+                    tokens[i - 2],
+                    "class",
+                    "task-list-item" + (" enabled" if not disable_checkboxes else ""),
+                )
+                attr_set(
+                    tokens[parent_token(tokens, i - 2)], "class", "contains-task-list"
+                )
+
+    md.core.ruler.after("inline", "github-tasklists", fcn)
+
+    def attr_set(token, name, value):
+        index = token.attrIndex(name)
+        attr = [name, value]
+        if index < 0:
+            token.attrPush(attr)
+        else:
+            token.attrs[index] = attr
+
+    def parent_token(tokens, index):
+        target_level = tokens[index].level - 1
+        for i in range(1, index + 1):
+            if tokens[index - i].level == target_level:
+                return index - i
+        return -1
+
+    def is_todo_item(tokens, index):
+        return (
+            is_inline(tokens[index])
+            and is_paragraph(tokens[index - 1])
+            and is_list_item(tokens[index - 2])
+            and starts_with_todo_markdown(tokens[index])
+        )
+
+    def todoify(token: Token, token_constructor):
+        token.children.insert(0, make_checkbox(token, token_constructor))
+        token.children[1].content = token.children[1].content[3:]
+        token.content = token.content[3:]
+
+        if use_label_wrapper:
+            if use_label_after:
+                token.children.pop()
+
+                # Use large random number as id property of the checkbox.
```

Comment:
just a nitpick, but really it would be better to use https://docs.python.org/3/library/uuid.html here, e.g.:

```python
from uuid import uuid4

checklist_id = f"task-item-{uuid4()}"
```

Reply:
I agree, and I had the same thought but chose to port the plugin ”as is”.

Reply:
I added the change and a comment in port.yml